pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -767,6 +767,8 @@ func newUnschedulablePodsMap(metricRecorder metrics.MetricRecorder) *Unschedulab
 // may be different than what scheduler has here. We should be able to find pods
 // by their UID and update/delete them.
 type nominatedPodMap struct {
+	// podAlive is a function to verify if the given pod is alive.
+	podAlive func(pod *v1.Pod) bool
```

Comment:
Use a function over a field (like podLister), so tests that don't care about this can simply inject a dummy alwaysTrue function.